### ISSUE TITLE
Can assign types even though interface field overriden using Omit

### ISSUE THREAD

**Issue by tom-thorne** [microsoft/TypeScript#36604]
**TypeScript Version:**  3.7.5

**Search Terms:**

Omit field override intersection assign missing error

**Expected behavior:**

If I use Omit and an intersection to override a field within an interface, when I assign an object of new type to original type I am expecting that typescript would give an error.

For example, in the example below, I expect to see a compile error in the line ` x = v; `
But unfortunately there isn't.

However, assigning the field explicitly between the 2 types does give an error.

**Playground Link:** [Provided](https://www.typescriptlang.org/play/#code/LAKAlgdgLgpgTgMwIYGMYAIAqB7dBvUddBMGAGwBMB+ALnQgFcBbAI3gG5D0kJsoALeADFSlOgGcocSAHNOIAL6hQUAJ4AHDELjYm6ALxE8xURTqMyZdAvQAydAHkmYKAB4cAGnQByEuQreAHzKIAgMEChQYNgQxNjYABQAbnTaugCU+FxkMFDoAB50OPJE+QboSexEAPTV6DD5mpHoULj8SEkYPPVwOnDoLAx5AvDoYOIQ3lBc+QB0fpTlSfOmVbVYGhjeFmTeY+L0fNzi4mAyEEgsOS24apo+jKyjAD7o4RQwJBAwAaBKIKAgA)

**Actual behavior:**

**Code**
```ts

interface To {
  field?: number;
  anotherField: string;
}

type From =  { field: null } & Omit<To, 'field'>

function foo(v: From) {
  let x: To;
  x = v;  // expect to have an error but there isn't
  x.field = v.field; // Type 'null' is not assignable to type 'number | undefined'
}


```

<details><summary><b>Compiler Options</b></summary>

```json
{
  "compilerOptions": {
    "noImplicitAny": true,
    "strictNullChecks": true,
    "strictFunctionTypes": true,
    "strictPropertyInitialization": true,
    "strictBindCallApply": true,
    "noImplicitThis": true,
    "noImplicitReturns": true,
    "useDefineForClassFields": false,
    "alwaysStrict": true,
    "allowUnreachableCode": false,
    "allowUnusedLabels": false,
    "downlevelIteration": false,
    "noEmitHelpers": false,
    "noLib": false,
    "noStrictGenericChecks": false,
    "noUnusedLocals": false,
    "noUnusedParameters": false,
    "esModuleInterop": true,
    "preserveConstEnums": false,
    "removeComments": false,
    "skipLibCheck": false,
    "checkJs": false,
    "allowJs": false,
    "declaration": true,
    "experimentalDecorators": false,
    "emitDecoratorMetadata": false,
    "target": "ES2017",
    "module": "ESNext"
  }
}
```

</details>

**Comment by ilogico** [comment]
It's funny, you can get the correct behavior with:
```ts
interface From extends Omit<To, 'field'> {
  field: null;
}
```

**Comment by ekilah** [comment]
thanks for this, just ran into it as well. [playground link](https://www.typescriptlang.org/play/#code/C4TwDgpgBAKjBOBXCB5eAxAhgGwM7QF4oByYJCYqAHxIDMd9iAoJgelagHdoALTAN2iYotCJxEBLCNgAmuLtE6YAdsCjAA9lDDwNAYwi5cAOigBlDQFsh8aMsSWARhHi4ANFFxWb0PTwh6ANaOGgAeALTYEoHQ-DjIuGwcwHxq3FC2OoYQqlwSKVAABnDkaFh4EIXGTKCQsOga8JYAIpjAwkQA3kxQUCCGaAByGgCKCcASGsoAXLAIyGUMED0ithC0jZZjhhNTs-ZOLkwAvizsnpjWUDEg8pjyMA1Nre0ejohqelOC8GolCxgluotMIQhpsFAJMpcMAIJgZB4IMYAOamDbwKDCABSZhQgygAEEAAoASSS6n8UDB2DhykhdzwWg0YF2yhwkLpKWgxJJxHktgAjuMqRoZCAauBoDAzIhHJZ8k9LDANASwBIoF0Vv1cENRuNJsoAPyzam0la0Nborb6vZQA7OeAnM7JSm0RDKPSshTEGRQPQ4CGcfI8Cnc0lQGRtTBMN0er24WXy4AACjAmHgl1ws2liYVm2VqokAEpZkTdPL8AAeFQgAB8UG6vVswEQ8DpZasEnwxlsXmwgmTRadTC+0LUVpemGzisnGobWoG8GG2xhBtmpHIxDc5stmxXrNmABYAEzb07k+5QCChSCeiAIildkSYCR4KkBTCIfAKESNZEaNRND9b4XDUQoN2QYhCmBIoyGQQomATOV8mTCcoyHckaWAPlPAgaAJFoCk2k8CZsEDRpAjuWhYQxSxMECKFkSgDRW1-Jp1ElPIChUKAADIORhOEZGmCU6keTZJ2POdOlWPCrX3Nc7QcB0oGOPiGz6RdlxtGY5lKQEKlUkcphhNiWijY9pwkizpIXHUlz1HZFIgiht16C05L3HSj1PYdzmULQXF0eBjGMABCRDcxTND2mPDDaildBdyaFBlGwEBpNk9YvKc217RcIymChGj6AMKASRnKMAGYr1CWFlDkepkssVL0vnXptV1BTbX+VADPwYdR1MmLMCq2YKus9oas1DqtMc1dbRcrcd08pput0k8z2dTF5B4YBgDALN2GRYNZWML5LFYeU9F0LxqNYGBJTMG6JBZVguwTQxWCqgA2H6AAZDwAYg+5ALusVRwgAVgADgARgATh+hHD1+zxEGRZEdncckv0YzE6WKlxSvwschLyMjJFCR9EiQpNUMm0aMPOUc3VwRj0o8EAWL9HiuTpOiYkxWDJWgi0rFDIoifgEnCg8FRfX5zFKafbhKcwZEX2UUTEsZw85wm55qpYIbxz1qyjfafWZs0+ztNy3SlrcrL5O8qBNuHOmUJGw8hyAA)

seems to be solved by switching to the `Nightly` build of TS which I assume includes the changes in the 3.9 beta, which claims to have this fixed via #37195 

seems like one step closer to fixing gaps between types and interfaces, like I brought up in #24970
